server/mvcc/backend/backend.go

```diff
@@ -221,10 +238,45 @@ func (b *backend) ConcurrentReadTx() ReadTx {
 	defer b.readTx.RUnlock()
 	// prevent boltdb read Tx from been rolled back until store read Tx is done. Needs to be called when holding readTx.RLock().
 	b.readTx.txWg.Add(1)
+
```

Comment:
I think lines below can be simplified as:

```go
// inspect/update cache recency iff there's no ongoing update to the cache
// this falls through if there's no cache update
b.txReadBufferCache.mu.Lock()

curCache := b.txReadBufferCache.buf
curCacheVer := b.txReadBufferCache.bufVersion
curBufVer := b.readTx.buf.bufVersion

emptyCache := curCache == nil
staleCache := curCacheVer != curBufVer

var buf *txReadBuffer
switch {
  case emptyCache:
    // perform safe copy of buffer while holding "b.txReadBufferCache.mu.Lock"
    curBuf := b.readTx.buf.unsafeCopy()
    buf = &curBuf

  case staleCache:
    // to maximize the concurrency, try unsafe copy of buffer
    // release the lock while copying buffer -- cache may become stale again
    b.txReadBufferCache.mu.Unlock()
    curBuf := b.readTx.buf.unsafeCopy()
    b.txReadBufferCache.mu.Lock()
    buf = &curBuf

  default:
    // neither empty nor stale cache, just use the current buffer
    buf = curCache
}
if emptyCache || curCacheVer == b.readTx.buf.bufVersion {
    b.txReadBufferCache.buf = &curBuf
    b.txReadBufferCache.bufVersion = curBufVer
}

b.txReadBufferCache.mu.Unlock()
```

Please let me know if I understood your change correctly.

Reply:
in this line:
`if emptyCache || curCacheVer == b.readTx.buf.bufVersion {`
should it be:
`curCacheVer == b.txReadBufferCache.bufVersion` ?